node/impl/full/eth.go

```diff
@@ -840,6 +845,12 @@ func (a *EthModule) EthSendRawTransaction(ctx context.Context, rawTx ethtypes.Et
 		return ethtypes.EmptyEthHash, err
 	}
 
+	// make it immediately available in the transaction hash lookup db, even though it will also
+	// eventually get there via the mpool
+	if err := a.EthTxHashManager.TransactionHashLookup.UpsertHash(txHash, smsg.Cid()); err != nil {
+		return ethtypes.EmptyEthHash, err
```

Comment:
I don't think we should return an error just because the upsert fails. Otherwise the client will wrongly assume that the txn hasn't been sent to the mpool. Best to just log an error here and return.

Reply:
OK, it's a bit of a no-mans land if it fails but at least there'll be 2 more chances.

Reply:
Yeah and it's not great to tell clients their tx submission failed but then go around and execute it later.